File: backend/app/schemas/document.py
"""Document schemas."""

import enum
import uuid
from datetime import datetime

from pydantic import BaseModel


class DocType(enum.StrEnum):
    """Document type enum."""

    RESUME = "resume"
    JOB_DESCRIPTION = "job_description"


class DocumentResponse(BaseModel):
    """Document response schema."""

    id: uuid.UUID
    name: str
    doc_type: DocType
    mime_type: str
    file_size: int
    extracted_text: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/document.ts
/** Document type enum. */
export type DocType = "resume" | "job_description";

/** Document object returned from the API. */
export interface Document {
  id: string;
  name: string;
  doc_type: DocType;
  mime_type: string;
  file_size: number;
  extracted_text: string | null;
  created_at: string;
  updated_at: string;
}
